### ISSUE TITLE
Too few arguments error in media manager

### ISSUE THREAD

**Issue by n9iels** [joomla/joomla-cms#17129]
### Steps to reproduce the issue
1. Create `onContentBeforeDisplay` plugin event without a fallback variable for the `$page` parameter. Like this:
```php
public function onContentBeforeDisplay($context, &$row, &$params, $page) {
}
```
You can also temporary remove the `= 0` after the `$page` for a existing plugin. For example the vote plugin on this line: https://github.com/joomla/joomla-cms/blob/staging/plugins/content/vote/vote.php#L62
2. Set the error reporting to `maximum`
3. Go to the media manager and notice the following error:
```
Too few arguments to function PlgContentVote::onContentBeforeDisplay(), 3 passed in /usr/local/apache2/htdocs/libraries/joomla/event/event.php on line 70 and exactly 4 expected
```
### Expected result
No errors or warning

### Actual result
```
Too few arguments to function PlgContentVote::onContentBeforeDisplay(), 3 passed in /usr/local/apache2/htdocs/libraries/joomla/event/event.php on line 70 and exactly 4 expected
```

### System information (as much as possible)
PHP 7.1.7
Latest Joomla! staging

### Additional comments
The issue also appear in a older PHP version (PHP7.x). But it seems like that the latest php version raises a error with higher priority than previous versions.

The error is caused by multiple event dipatchers in the `administrator/components/com_media/views/medialist/tmpl` folder that call the `onContentBeforeDisplay` event without the the fourth argument, `$page`. A solution might be adding a `0` as fourth parameter. However I'm not sure if this is the right solution.

**Comment by zero-24** [comment]
hmm IMO we could just fix that in the core but any 3rd party that is triggering this event will throw that error too. This is what the default value is for. ;) For sure you can add the default (0) to the calling events where not done yet. But this would not fix the 3rd Party triggerd events.

**Comment by n9iels** [comment]
I created a PR for this. I think that fixing for 3rd party extensions is very hard. We cannot change the event dispatcher specific for this event.
The default value is indeed exact for this situation :-)

**Comment by ghost** [comment]
closed as having PR #17133<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/17129">issues.joomla.org/tracker/joomla-cms/17129</a>.</sub>
